presto-spi/src/main/java/com/facebook/presto/spi/type/LongDecimalType.java

```diff
@@ -88,18 +88,21 @@ public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int
     @Override
     public long hash(Block block, int position)
     {
-        long low = block.getLong(position, 0);
-        long high = block.getLong(position, SIZE_OF_LONG);
```

Comment:
This is the only type we need `offset`

Reply:
This could have some performance penalty for `joins` and `group by` on decimal type. I am not sure how often this is used though. I am kind of on the edge if this should stay and if we should performance benchmark it. What do you think? Some third opinion, @losipiuk ?

Reply:
@sopel39 : Discussed with @haozhun , since creating a Slice is expensive and we want to avoid it (that's why `offset` is introduced in `getX`). We will just keep the `offset` for now.

Reply:
Yup. In the long term it would be great if we can have structural block for decimals so we could avoid `Slice` representation whatsoever.

Reply:
@sopel39 : But still needs to return a reference right? :(